src/cpp/session/modules/SessionProfiler.R

```diff
@@ -13,4 +13,54 @@
 #
 #
 
+.rs.addJsonRpcHandler("start_profiling", function(profilerOptions)
+{
+   tryCatch({
+      tempPath <- .Call("rs_profilesPath")
+      if (!dir.exists(tempPath)) {
+         dir.create(tempPath, recursive = TRUE)
+      }
+      
+   	fileName <- tempfile(fileext = ".rprof", tmpdir = tempPath)
 
+      Rprof(filename = fileName, line.profiling = TRUE)
```

Comment:
Good point. I'm now adding a hook into Rprof directly to be able to detect correctly start/stop events.

BTW. It's not my intention to make the UI fully interoperable with the console; but we could certainly do more work here if we are interested. My take: For those that prefer to use the console, you might also prefer to open the profile pane programmatically (by opening the `.rprof` file yourself).